rts/motoko-rts/src/stabilization/layout/stable_blob.rs

```diff
@@ -0,0 +1,92 @@
+use crate::{
+    memory::{alloc_blob, Memory},
+    stabilization::{
+        deserialization::stable_memory_access::StableMemoryAccess,
+        serialization::{
+            stable_memory_stream::{ScanStream, StableMemoryStream, WriteStream},
+            SerializationContext,
+        },
+    },
+    types::{size_of, Blob, Bytes, Value, TAG_BLOB},
+};
+
+use super::{
+    round_to_u64, write_padding_u64, Serializer, StableToSpace, StableValue, StaticScanner,
+};
+
+#[repr(C)]
+#[derive(Default)]
+pub struct StableBlob {
+    byte_length: u64,
+    // Dynamically sized body with `byte_length` bytes. No pointers to be scanned.
+    // Zero padding to align to next `u64`.
+    // Note: The rounding of object sizes to at least 2 bytes is necessary for the skewed pointer representation.
+}
+
+impl StableBlob {
+    pub fn byte_length(&self) -> u64 {
+        self.byte_length
+    }
+}
+
+impl StaticScanner<StableValue> for StableBlob {}
+
+impl Serializer<Blob> for StableBlob {
+    unsafe fn serialize_static_part(
+        _stable_memory: &mut StableMemoryStream,
+        main_object: *mut Blob,
+    ) -> Self {
+        StableBlob {
+            byte_length: main_object.len().as_usize() as u64,
+        }
+    }
+
+    unsafe fn serialize_dynamic_part(
+        stable_memory: &mut StableMemoryStream,
+        main_object: *mut Blob,
+    ) {
+        let byte_length = main_object.len().as_usize();
+        stable_memory.raw_write(main_object.payload_addr() as usize, byte_length);
```

Comment:
Like array serialization, this raw_write is non-incremental and I guess could fail to fit in the increment...

Reply:
I will check whether this is an issue in practice for large blobs. I tried to avoid the complexity of incremental loops, except for the necessary cases (which I have seen was the array scanning in Cheney's algorithm).

Reply:
I measured the serialization of a 2 GB blob: It takes about 2.7 billion instructions. I believe this is still practically small enough to fit into a stabilzation increment.